ChernitskayaED/lab_1/main.cpp

```diff
@@ -0,0 +1,213 @@
+        }
+    }
+    void push_back(const T& val);
+    void resize(unsigned size);
+    unsigned size();
+    void print();
+    bool insert(unsigned pos, const T& val);
+    bool contains(const T& val);
+    int find(const T& val);
+    T& operator [](unsigned index);
+    List& operator=(const List& clist); 
+    List(const List& clist);
+};
+
+List::List(const List& clist) {                        
+    Node* current = clist.head;
+    for (int i = 0; i < clist.size_now; i++) {
+        push_back(current->value);
+        current = current->next;
+    }
+}
+void List::resize(unsigned size)
+{
+    if (size_now == 0 && size != 0)
+        head = new Node();
+    Node* current = head;
+    if (size > size_now) {
+        for (int i = 0; i < size - 1; i++) {
+            if (size_now == 0 || (i >= size_now - 1))
+            {
+                current->next = new Node{ 0 };
+            }
+            current = current->next;
+        }
+    }
+    else {    
+        for (int i = 0; i < size_now - size; i++) {
+            current = head->next;
+            delete head;
+            head = current;
+        }
+    }
+    size_now = size;
+}
+
+void List::push_back(const T& val)
+{
+    if (head == nullptr) { 
+        head = new Node{ val };
+    }
+    else {                 
+        Node* current = this->head;
+        while (current->next != nullptr)
+            current = current->next;
+        current->next = new Node{ val };
+    }
+    size_now++;
+}
+
+unsigned List::size() {
+    return size_now;
+}
+
+bool List::insert(unsigned pos, const T& val) {
+    if (pos > size_now || pos < 0)
+        return false;
+    if (pos == 0) {
+        head = new Node{ val, head };
+        size_now++;
+        return true;
+    }
+    if (pos == size_now) {
+        push_back(val);
+        return true;
+    }
+    else {
+        Node* last = this->head;       
+        for (int i = 0; i < pos - 1;i++)   
+        {
+            last = last->next;
+        }
+        Node* newElement = new Node{ val, last->next }; 
+        last->next = newElement;  
+        size_now++;
+        return true;
+    }
+}
+
+bool List::contains(const T& val) {
+    Node* current = head;
+    for (int i = 0; i < size_now; i++) {
+        if (current->value == val)
+            return true;
+        current = current->next;
+    }
+    return false;
+}
+
+int List::find(const T& val)
+{
+    Node* current = head;
+    for (int i = 0; i < size_now; i++) {
+        if (current->value == val) {
+            return i;
+        }
+        current = current->next;
+    }
+    return -1;
+}
+
+T& List::operator [](unsigned index) {
+    if (index < 0 || index >= size_now) {
+        std::cout << "wrong type";
+        system("Pause");                   
+    }
+    Node* current = head;
+    for (int i = 0; i < index; i++) {
+        current = current->next;
+    }
+    return current->value;
+}
+
+List& List::operator=(const List& clist) {   
```

Comment:
It adds elements to back. resize(0) should be before, or resize(clist.size_now) better